lib/model.js

```diff
@@ -829,6 +829,12 @@ Model.prototype.deleteOne = function deleteOne(options) {
   query.post(function queryPostDeleteOne(cb) {
     self.constructor._middleware.execPost('deleteOne', self, [self], {}, cb);
   });
+  query.transform(function setIsDeleted(result) {
+    if (result && result.deletedCount != null && result.deletedCount > 0) {
```

Comment:
Nitpicky suggestion:
```suggestion
    if (result?.deletedCount > 0) {
```

Take it with a grain of salt for now, will be looking more into the benchmarks of optional chaining later on my PR 👍